panic: Search domain is not valid

<!-- This form is for bug reports and feature requests ONLY!
Also make sure that you visit our User Guide at https://kubevirt.io/user-guide/
-->
Hi please help to understand what's wrong here.
 
when I run the vm 
------------
apiVersion: kubevirt.io/v1alpha3
kind: VirtualMachine
metadata:
  name: testvm2
spec:
  running: false
  template:
    metadata:
      labels:
        kubevirt.io/size: small
        kubevirt.io/domain: testvm
    spec:
      domain:
        devices:
          disks:
            - name: containerdisk
              disk:
                bus: virtio
            - name: cloudinitdisk
              disk:
                bus: virtio
          interfaces:
          - name: default
            bridge: {}
        resources:
          requests:
            memory: 64M
      networks:
      - name: default
        pod: {}
      volumes:
        - name: containerdisk
          containerDisk:
            image: kubevirt/cirros-container-disk-demo
        - name: cloudinitdisk
          cloudInitNoCloud:
            userDataBase64: SGkuXG4=
--------

logs:
------------------
{"component":"virt-launcher","level":"info","msg":"Collected all requested hook sidecar sockets","pos":"manager.go:68","timestamp":"2020-11-11T14:15:52.927542Z"}
{"component":"virt-launcher","level":"info","msg":"Sorted all collected sidecar sockets per hook point based on their priority and name: map[]","pos":"manager.go:71","timestamp":"2020-11-11T14:15:52.927608Z"}
{"component":"virt-launcher","level":"info","msg":"Connecting to libvirt daemon: qemu:///system","pos":"libvirt.go:380","timestamp":"2020-11-11T14:15:52.933479Z"}
{"component":"virt-launcher","level":"info","msg":"Connecting to libvirt daemon failed: virError(Code=38, Domain=7, Message='Failed to connect socket to '/var/run/libvirt/libvirt-sock': No such file or directory')","pos":"libvirt.go:388","timestamp":"2020-11-11T14:15:52.934110Z"}
{"component":"virt-launcher","level":"info","msg":"libvirt version: 6.0.0, package: 16.fc31 (Unknown, 2020-04-07-15:55:55, )","subcomponent":"libvirt","thread":"71","timestamp":"2020-11-11T14:15:52.954000Z"}
{"component":"virt-launcher","level":"info","msg":"hostname: testvm2","subcomponent":"libvirt","thread":"71","timestamp":"2020-11-11T14:15:52.954000Z"}
{"component":"virt-launcher","level":"error","msg":"internal error: Child process (/usr/sbin/dmidecode -q -t 0,1,2,3,4,17) unexpected exit status 1: /dev/mem: No such file or directory","pos":"virCommandWait:2709","subcomponent":"libvirt","thread":"71","timestamp":"2020-11-11T14:15:52.954000Z"}
{"component":"virt-launcher","level":"info","msg":"Connected to libvirt daemon","pos":"libvirt.go:396","timestamp":"2020-11-11T14:15:53.435227Z"}
{"component":"virt-launcher","level":"info","msg":"Registered libvirt event notify callback","pos":"client.go:394","timestamp":"2020-11-11T14:15:53.437379Z"}
{"component":"virt-launcher","level":"info","msg":"Marked as ready","pos":"virt-launcher.go:71","timestamp":"2020-11-11T14:15:53.437494Z"}
[labuser@kvm-srv28 ~]$ ./kubectl logs   virt-launcher-testvm2-4269q compute
{"component":"virt-launcher","level":"info","msg":"Collected all requested hook sidecar sockets","pos":"manager.go:68","timestamp":"2020-11-11T14:15:52.927542Z"}
{"component":"virt-launcher","level":"info","msg":"Sorted all collected sidecar sockets per hook point based on their priority and name: map[]","pos":"manager.go:71","timestamp":"2020-11-11T14:15:52.927608Z"}
{"component":"virt-launcher","level":"info","msg":"Connecting to libvirt daemon: qemu:///system","pos":"libvirt.go:380","timestamp":"2020-11-11T14:15:52.933479Z"}
{"component":"virt-launcher","level":"info","msg":"Connecting to libvirt daemon failed: virError(Code=38, Domain=7, Message='Failed to connect socket to '/var/run/libvirt/libvirt-sock': No such file or directory')","pos":"libvirt.go:388","timestamp":"2020-11-11T14:15:52.934110Z"}
{"component":"virt-launcher","level":"info","msg":"libvirt version: 6.0.0, package: 16.fc31 (Unknown, 2020-04-07-15:55:55, )","subcomponent":"libvirt","thread":"71","timestamp":"2020-11-11T14:15:52.954000Z"}
{"component":"virt-launcher","level":"info","msg":"hostname: testvm2","subcomponent":"libvirt","thread":"71","timestamp":"2020-11-11T14:15:52.954000Z"}
{"component":"virt-launcher","level":"error","msg":"internal error: Child process (/usr/sbin/dmidecode -q -t 0,1,2,3,4,17) unexpected exit status 1: /dev/mem: No such file or directory","pos":"virCommandWait:2709","subcomponent":"libvirt","thread":"71","timestamp":"2020-11-11T14:15:52.954000Z"}
{"component":"virt-launcher","level":"info","msg":"Connected to libvirt daemon","pos":"libvirt.go:396","timestamp":"2020-11-11T14:15:53.435227Z"}
{"component":"virt-launcher","level":"info","msg":"Registered libvirt event notify callback","pos":"client.go:394","timestamp":"2020-11-11T14:15:53.437379Z"}
{"component":"virt-launcher","level":"info","msg":"Marked as ready","pos":"virt-launcher.go:71","timestamp":"2020-11-11T14:15:53.437494Z"}
{"component":"virt-launcher","level":"info","msg":"Hardware emulation device '/dev/kvm' not present. Using software emulation.","pos":"converter.go:699","timestamp":"2020-11-11T14:15:55.718368Z"}
{"component":"virt-launcher","level":"info","msg":"In-kernel virtio-net device emulation '/dev/vhost-net' not present. Falling back to QEMU userland emulation.","pos":"converter.go:712","timestamp":"2020-11-11T14:15:55.718421Z"}
{"component":"virt-launcher","kind":"","level":"info","msg":"Executing PreStartHook on VMI pod environment","name":"testvm2","namespace":"default","pos":"manager.go:982","timestamp":"2020-11-11T14:15:55.718782Z","uid":"1d6f8c5f-52bd-4cac-bf26-4fe70ea83f0a"}
{"component":"virt-launcher","kind":"","level":"info","msg":"Starting PreCloudInitIso hook","name":"testvm2","namespace":"default","pos":"manager.go:1001","timestamp":"2020-11-11T14:15:55.718814Z","uid":"1d6f8c5f-52bd-4cac-bf26-4fe70ea83f0a"}
{"component":"virt-launcher","kind":"","level":"info","msg":"bridge pod interface: VIF: { Name: eth0, IP: 10.42.0.224, Mask: ffffffff, MAC: 42:d3:2b:72:99:58, Gateway: 169.254.1.1, MTU: 1450, IPAMDisabled: false, TapDevice: tap0} \u0026{vmi:0xc000241200 vif:0xc000257400 iface:0xc000195680 virtIface:0xc000458680 podNicLink:\u003cnil\u003e domain:0xc0001b6500 podInterfaceName:eth0 bridgeInterfaceName:k6t-eth0}","name":"testvm2","namespace":"default","pos":"podinterface.go:431","timestamp":"2020-11-11T14:15:55.719233Z","uid":"1d6f8c5f-52bd-4cac-bf26-4fe70ea83f0a"}
{"component":"virt-launcher","level":"info","msg":"Found nameservers in /etc/resolv.conf: \n+\u0000\n","pos":"converter.go:1696","timestamp":"2020-11-11T14:15:55.719366Z"}
{"component":"virt-launcher","level":"info","msg":"Found search domains in /etc/resolv.conf: default.svc.cluster.local svc.cluster.local cluster.local HABANA-LABS.COM","pos":"converter.go:1697","timestamp":"2020-11-11T14:15:55.719382Z"}
{"component":"virt-launcher","level":"info","msg":"Starting SingleClientDHCPServer","pos":"dhcp.go:64","timestamp":"2020-11-11T14:15:55.719466Z"}
{"component":"virt-launcher","level":"error","msg":"failed to run DHCP: Search domain is not valid: 'HABANA-LABS.COM'","pos":"common.go:358","timestamp":"2020-11-11T14:15:55.719734Z"}
panic: Search domain is not valid: 'HABANA-LABS.COM'

goroutine 220 [running]:
kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/network.(*NetworkUtilsHandler).StartDHCP.func1(0xc000257400, 0xc000f08150, 0x8, 0xc000603aa0, 0xc0001c65c0, 0x1, 0x1, 0xc001a065c0, 0x4, 0x4, ...)
        pkg/virt-launcher/virtwrap/network/common.go:359 +0x372
created by kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/network.(*NetworkUtilsHandler).StartDHCP
        pkg/virt-launcher/virtwrap/network/common.go:344 +0x301
{"component":"virt-launcher","level":"error","msg":"dirty virt-launcher shutdown","pos":"virt-launcher.go:508","reason":"exit status 2","timestamp":"2020-11-11T14:15:55.725449Z"}

-------------------

**Anything else we need to know?**:

**Environment**:
./kubectl version
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.8", GitCommit:"9f2892aab98fe339f3bd70e3c470144299398ace", GitTreeState:"clean", BuildDate:"2020-08-13T16:12:48Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.10", GitCommit:"62876fc6d93e891aa7fbe19771e6a6c03773b0f7", GitTreeState:"clean", BuildDate:"2020-10-15T01:43:56Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}


./virtctl version
Client Version: version.Info{GitVersion:"v0.35.0", GitCommit:"b77b858ac9345ae6858cf409e1833f4548ec2809", GitTreeState:"clean", BuildDate:"2020-11-09T13:38:36Z", GoVersion:"go1.13.14", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{GitVersion:"{gitVersion}", GitCommit:"{gitCommit}", GitTreeState:"{gitTreeState}", BuildDate:"{buildDate}", GoVersion:"go1.13.14", Compiler:"gc", Platform:"linux/amd64"}


cat /etc/os-release
NAME="CentOS Linux"
VERSION="7 (Core)"

Linux 3.10.0-1062.9.1.el7.x86_64 #1 SMP Fri Dec 6 15:49:49 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux

- KubeVirt version (use `virtctl version`):
- Kubernetes version (use `kubectl version`):
- VM or VMI specifications:
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

--- a/pkg/util/net/dns/resolveconf.go
+++ b/pkg/util/net/dns/resolveconf.go
@@ -56,7 +56,8 @@ func ParseSearchDomains(content string) ([]string, error) {
 		if strings.HasPrefix(line, domainSearchPrefix) {
 			doms := strings.Fields(strings.TrimPrefix(line, domainSearchPrefix))
 			for _, dom := range doms {
-				searchDomains = append(searchDomains, dom)
+				// domain names are case insensitive but kubernetes allows only lower-case
+				searchDomains = append(searchDomains, strings.ToLower(dom))
 			}
 		}
 	}

--- a/pkg/util/net/dns/resolveconf_test.go
+++ b/pkg/util/net/dns/resolveconf_test.go
@@ -69,5 +69,12 @@ var _ = Describe("Resolveconf", func() {
 			Expect(searchDomains).To(Equal([]string{"local"}))
 			Expect(err).To(BeNil())
 		})
+
+		It("should normalize search domains to lower-case", func() {
+			resolvConf := "search LoCaL\nnameserver 8.8.8.8\n"
+			searchDomains, err := ParseSearchDomains(resolvConf)
+			Expect(searchDomains).To(Equal([]string{"local"}))
+			Expect(err).To(BeNil())
+		})
 	})
 })

